### ISSUE TITLE
Should we use setTimeout or transitionend to hide upload complete icon?

### ISSUE THREAD

**Issue by oskarwrobel** [ckeditor/ckeditor5-image#206]
Follow up https://github.com/ckeditor/ckeditor5-image/issues/204.

Image upload will have complete icon soon. This icon appears after succeed upload and hides after certain amount of time. Current implementation uses `setTimoeut` to remove icon after 3 seconds: https://github.com/ckeditor/ckeditor5-image/blob/d74552f7249f3c157a2a7162e4ffbb2b8e8da5ef/src/imageupload/imageuploadprogress.js#L214-L216 With @dkonopka we are thinking about using `transitionend` listener instead.

### pros
- Animation time is completly controlled by CSS and doesn't have to be synchronise with JS

### cons
- Icon element needs to have animation otherwise won't be removed from DOM. This is important when someone will wan't to override icon styles.

**Comment by oleq** [comment]
I'd rather see it handled by CSS if possible. Can you tell us more about the "cons" because I'm not sure it's an obvious situation?

**Comment by oskarwrobel** [comment]
If we decide to remove icon after animation (using `transitionend` event) then animation will be required for this icon element. When someone will override CSS and remove transition from this element then `transitionend` event will be never fired and this element won't be removed from DOM.

**Comment by oleq** [comment]
I don't think this is an issue. If developers decide they want a different UI, they may need a different JS supporting it too.
